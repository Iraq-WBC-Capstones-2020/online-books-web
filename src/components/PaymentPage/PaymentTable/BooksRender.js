import React from 'react';
export default function BooksRender({ books, changeCheck }) {
  return books.map((book, index) => {
    const { title, copies, price, id, checked } = book;

    return (
      <tr
        key={index}
        className="text-center"
        style={{ textDecoration: !checked ? 'line-through' : 'none' }}
      >
        <td>
          <input
            type="checkbox"
            onChange={(event) => changeCheck(id, event.target.checked)}
            checked={checked}
            line-through
          />
        </td>
        <td>{title}</td>
        <td>{copies}</td>
        <td>{price}</td>
      </tr>
    );
  });
}
