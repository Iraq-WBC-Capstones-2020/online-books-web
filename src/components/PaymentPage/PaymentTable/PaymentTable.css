.table thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #2f80ed;
}

.table td,
.table th {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #2f80ed;
}
/*# sourceMappingURL=PaymentTable.css.map */
