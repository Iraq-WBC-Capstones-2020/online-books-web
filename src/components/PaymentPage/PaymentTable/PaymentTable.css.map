{
    "version": 3,
    "mappings": "AAAA,AACE,MADI,CACJ,KAAK,CAAC,EAAE,CAAC;EACP,cAAc,EAAE,MAAM;EACtB,aAAa,EAAE,iBAAiB;CACjC;;AAJH,AAKE,MALI,CAKJ,EAAE;AALJ,MAAM,CAMJ,EAAE,CAAC;EACD,OAAO,EAAE,OAAO;EAChB,cAAc,EAAE,GAAG;EACnB,UAAU,EAAE,iBAAiB;CAC9B",
    "sources": [
        "PaymentTable.scss"
    ],
    "names": [],
    "file": "PaymentTable.css"
}