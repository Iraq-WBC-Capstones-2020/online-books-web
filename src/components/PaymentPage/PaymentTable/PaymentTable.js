import React from 'react';
import './PaymentTable.scss';
import { useTranslation } from 'react-i18next';
import BooksRender from './BooksRender';

import {
  MDBContainer,
  MDBRow,
  MDBCol,
  MDBTable,
  MDBTableBody,
  MDBTableHead,
} from 'mdbreact';

const NUMBER_OF_BOOKS = 4;

const BOOKS = Array(NUMBER_OF_BOOKS).fill({
  title: 'book',
  copies: 1,
  price: '15 $',
  id: 3,
});

export default function PaymentTable() {
  const { t } = useTranslation();
  return (
    <MDBContainer>
      <MDBRow>
        <MDBCol md="12">
          <div className="text-center">
            <p className="h5 mt-5 text-gray-800">
              {t('PaymentPage.booksOrder')}
            </p>
            <hr className="border-4 my-3 mx-4 bg-primary"></hr>
          </div>
          <MDBTable>
            <MDBTableHead>
              <tr>
                <th></th>
                <th>{t('PaymentPage.books')}</th>
                <th>{t('PaymentPage.copies')}</th>
                <th>{t('PaymentPage.price')}</th>
              </tr>
            </MDBTableHead>
            <MDBTableBody>
              <BooksRender books={BOOKS} />
              <tr>
                <td></td>
                <td>
                  <p className="h6">{t('PaymentPage.total')}</p>
                </td>
                <td>total copies here</td>
                <td>total price here</td>
              </tr>
            </MDBTableBody>
          </MDBTable>
        </MDBCol>
      </MDBRow>
    </MDBContainer>
  );
}
