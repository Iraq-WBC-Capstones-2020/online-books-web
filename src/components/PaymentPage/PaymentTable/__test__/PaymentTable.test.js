import React from 'react';
import renderer from 'react-test-renderer';
import PaymentTable from './../PaymentTable';

test('PaymentTable snapshot', () => {
  const tree = renderer.create(<PaymentTable />).toJSON();
  expect(tree).toMatchSnapshot();
});
