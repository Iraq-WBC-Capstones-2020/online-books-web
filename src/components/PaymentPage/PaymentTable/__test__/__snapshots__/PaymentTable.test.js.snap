// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentTable snapshot 1`] = `
<div
  className="container"
  data-test="container"
>
  <div
    className="row"
    data-test="row"
  >
    <div
      className="col-md-12"
      data-test="col"
    >
      <div
        className="text-center"
      >
        <p
          className="h5 mt-5 text-gray-800"
        >
          paymentPage.booksOrder
        </p>
        <hr
          className="border-4 my-3 mx-4 bg-primary"
        />
      </div>
      <div
        className=""
        data-test="table"
        style={
          Object {
            "maxHeight": undefined,
          }
        }
      >
        <table
          className="table"
        >
          <thead
            className=""
            data-test="table-head"
          >
            <tr
              className="text-center"
            >
              <th />
              <th>
                paymentPage.books
              </th>
              <th>
                paymentPage.copies
              </th>
              <th>
                paymentPage.price
              </th>
            </tr>
          </thead>
          <tbody
            data-test="table-body"
          >
            <tr
              className="text-center"
              style={
                Object {
                  "textDecoration": "line-through",
                }
              }
            >
              <td>
                <input
                  checked={false}
                  line-through={true}
                  onChange={[Function]}
                  type="checkbox"
                />
              </td>
              <td>
                book
              </td>
              <td>
                1
              </td>
              <td>
                15
              </td>
            </tr>
            <tr
              className="text-center"
              style={
                Object {
                  "textDecoration": "line-through",
                }
              }
            >
              <td>
                <input
                  checked={false}
                  line-through={true}
                  onChange={[Function]}
                  type="checkbox"
                />
              </td>
              <td>
                book
              </td>
              <td>
                1
              </td>
              <td>
                15
              </td>
            </tr>
            <tr
              className="text-center"
              style={
                Object {
                  "textDecoration": "line-through",
                }
              }
            >
              <td>
                <input
                  checked={false}
                  line-through={true}
                  onChange={[Function]}
                  type="checkbox"
                />
              </td>
              <td>
                book
              </td>
              <td>
                1
              </td>
              <td>
                15
              </td>
            </tr>
            <tr
              className="text-center"
              style={
                Object {
                  "textDecoration": "line-through",
                }
              }
            >
              <td>
                <input
                  checked={false}
                  line-through={true}
                  onChange={[Function]}
                  type="checkbox"
                />
              </td>
              <td>
                book
              </td>
              <td>
                1
              </td>
              <td>
                15
              </td>
            </tr>
            <tr>
              <td />
              <td>
                <p
                  className="h6 text-center"
                >
                  paymentPage.total
                </p>
              </td>
              <td
                  className="text-center"
              >
                0
              </td>
              <td
                  className="text-center"
              >
                0
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
